lib/http.js

```diff
@@ -973,6 +973,7 @@ OutgoingRequest.prototype = Object.create(OutgoingMessage.prototype, { construct
 
 OutgoingRequest.prototype._start = function _start(stream, options) {
   this.stream = stream;
+  this.options = options;
```

Comment:
I suppose that this has more to do with using more generic event handlers.  The options for request can contain additional options which would not be copied to the request object.